virtool/auth/mongo.py

```diff
@@ -0,0 +1,28 @@
+from virtool_core.models.enums import Permission
+
+from virtool.api.response import NotFound
+from virtool.mongo.core import DB
+
+
+async def check_in_mongo(mongo: DB, user_id: str, permission: Permission) -> bool:
+    """
+    Check permissions in Mongo.
+    """
+
+    return (
+        await mongo.users.count_documents(
+            {"_id": user_id, f"permissions.{permission.name}": True}
```

Comment:
This returns a count as soon as one document is found instead of traversing all documents.
```suggestion
            {"_id": user_id, f"permissions.{permission.name}": True}, limit=1
```